lib/semantics/scope.h

```diff
@@ -122,6 +138,8 @@ class Scope {
     return symbols_.emplace(name, &symbol);
   }
 
+  const std::list<EquivalenceSet> &equivalenceSets() const;
```

Comment:
Or `map<const Symbol *, EquivalenceSet *>`.

Reply:
A symbol may be in more than one equivalence set.

Reply:
Oh, I get it, you're not merging them as you go along.  Ok.